### DISCUSSION TITLE
[QUESTION] Get chat id when stating bot

### DISCUSSION THREAD

**Discussion by Ilocans** [python-telegram-bot/python-telegram-bot#3232] [Q&A]
### Issue I am facing

Hello,

I am starting to play with the python-telegram-bot and I was wondering if it is possible to either automaticaly start a command at the initalization (like if someone execute a /start)  in my main
or to get the chat_id ?

### Related part of your code

```python
def main():
    updater = Updater(token=Token, defaults=Defaults(parse_mode=ParseMode.HTML))
    dispatcher = updater.dispatcher
       
    dispatcher.add_handler(CommandHandler('start', startCommand)) # Accessed via /start
    updater.start_polling() # Start the bot
    
    # I would like to get the chat id here or execute the /start from which I can have the chat_id

    updater.idle() 

if __name__ == '__main__':
    main()
```

### Operating System

Windows

### Version of Python, python-telegram-bot & dependencies

```shell
Pythom 3.8
```

**Comment by Bibo-Joshi** [comment]
Hi. Get the chat_id of which chat? you can call whatever functions you like in the main, including bot methods. To send a message in `main`, you can e.g. use `updater.bot.send_message(…)`.

**Comment by Ilocans** [comment]
> Hi. Get the chat_id of which chat? you can call whatever functions you like in the main, including bot methods. To send a message in `main`, you can e.g. use `updater.bot.send_message(…)`.

Thanks for your reply.
Correct me if I am wrong but `updater.bot.send_message(…)` requires at least 2 arguments: a chat_id and a text, no ?

**Reply by Bibo-Joshi** [reply]
Yes, that's correct. What do you mean by "__the__ chat_id" though? Each chat has a different id, i.e. each private chat, each group chat and each channel.

**Reply by Ilocans** [reply]
Maybe it is a problem with my understanding on how this work, which is for now I guess limited.
But for me when I run the code above it launch a bot which I can chat, send command and it can respond or do some actions.

My objective is to automate some actions when the bot start without having to launch a command and to get a feedback from the bot in the chat I have with him. But it always seems to require a chat_id to send a message and the only way I have to get this chat_id seems to have first an interaction with the bot with a /command. Is there any other way to achieve what I want ?

thank you for your help,

**Reply by Bibo-Joshi** [reply]
There are several ways to get the id of a specific chat, but yes, you'll first have to obtain thet id.

The easiest way to get the chat_id is to read it from an incoming update, e.g. printing `update.effective_chat/user.id`. There are also possibilities to read the id from some Telgeram clients. If you want to know your own chat id, you can also send a message to `@JsonDumpsBot` and read it from the reply.
